### DISCUSSION TITLE
execute binaries on workers, but not installed on workers?

### DISCUSSION THREAD

**Discussion by dynabler** [jhuckaby/Cronicle#741] [Q&A] [answered]
I have one main server, which runs Scrapy via Cronicle. I added a worker without Scrapy, but this is not going to work.

My question is a bit vague, because I have no idea what to look for.

How do I allow main server to execute Scrapy on workers? If I install Scrapy on the worker, then I have to specify the Worker in Events. This defeats the purpose of having Cronicle in the first place, which is allowing it to divide the load as it sees fit among workers.

I also read this, not sure if that's the way to go: https://github.com/jhuckaby/Cronicle/issues/720#issuecomment-1940684792

Any direction where I'm supposed to look is welcome.

**Comment by jhuckaby** [comment, marked as answer]
I'm having trouble understanding your problem.

> I added a worker without Scrapy, but this is not going to work.

You will need to preinstall "Scrapy" on all your worker servers.  Cronicle cannot do this for you.

> This defeats the purpose of having Cronicle in the first place, which is allowing it to divide the load as it sees fit among workers.

To load balance between multiple worker servers, create a [Server Group](https://github.com/jhuckaby/Cronicle/blob/master/docs/WebUI.md#server-groups) that targets ALL of your worker servers.  You can do this by entering a hostname match pattern.

Then, when configuring your event, target the server **GROUP**, not an individual server.  You can then select a load balancing algorithm:

![Screenshot 2024-04-12 at 10 54 16 PM](https://github.com/jhuckaby/Cronicle/assets/700765/ce4d15f0-41e3-4f67-899a-4b6202fb4a50)

Good luck.

**Reply by dynabler** [reply]
Have been using Cronicle for a while now and what a time saver. I even stop with anything I'm doing, just to watch the home page fire up events. That for sure is far more exciting than Netflix. :-)
